fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -588,6 +594,43 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
             throw SearchExceptionUtil.buildNewInvalidSearchException(
                     "_sort search result parameter not supported with _include or _revinclude.");
         }
+        HashSet<String> resourceTypes = new HashSet<String>(); 
+        if (Resource.class.equals(resourceType)) {
+            // Because _include and _revinclude searches all require certain resource type modifier in 
+            // search parameter, so we just don't support it.
+            if (queryParameters.containsKey(SearchConstants.INCLUDE)
+                        || queryParameters.containsKey(SearchConstants.REVINCLUDE)) {
+                throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        "system search not supported with _include or _revinclude.");
+            }
+            
+            if (queryParameters.containsKey(SearchConstants.RESTYPE)) {
+                for (String ResTypes : queryParameters.get(SearchConstants.RESTYPE)) {
+                    List<String> tmpResourceTypes = Arrays.asList(ResTypes.split("\\s*,\\s*"));
+                    for (String resType : tmpResourceTypes) {
+                        if (ModelSupport.isResourceType(resType)) {
+                            resourceTypes.add(resType);
+                        } else {
+                            if (lenient) {
+                                log.log(Level.FINE, resType + " is not valid resource type!");
+                                continue;
+                            } else {
+                                throw SearchExceptionUtil.buildNewInvalidSearchException(
+                                        "_type search parameter has invilid resource type:" + resType);
+                            }
+                        }
+                    }
+                }
+            } 
+        }
+        
+        queryParameters.remove(SearchConstants.RESTYPE);
```

Comment:
How does the self uri look on the way back in the response body? is it still right?

Reply:
yeah, it's correct, because I added it back when building the url.
It's removed here to avoid the unknown search parameter error when processing the parameters.